src/lib/transaction_snark/test/account_timing/account_timing.ml

```diff
@@ -1643,4 +1643,162 @@ let%test_module "account timing check" =
+                      ; vesting_period = Mina_numbers.Global_slot.of_int 1
+                      ; vesting_increment = Currency.Amount.of_int 100_000
+                      }
+                  else Account_timing.Untimed ) )
+            |> Array.of_list
+          in
+          let sender_keypair = List.hd_exn keypairs in
+          let zkapp_keypair = List.nth_exn keypairs 1 in
+          let (update_timing_spec : Transaction_snark.For_tests.Spec.t) =
+            { sender = (sender_keypair, Account.Nonce.zero)
+            ; fee = Currency.Fee.of_int 1_000_000
+            ; fee_payer = None
+            ; receivers = []
+            ; amount = Currency.Amount.zero
+            ; zkapp_account_keypairs = [ zkapp_keypair ]
+            ; memo =
+                Signed_command_memo.create_from_string_exn "zkApp update timing"
+            ; new_zkapp_account = false
+            ; snapp_update =
+                (let timing =
+                   Zkapp_basic.Set_or_keep.Set
+                     ( { initial_minimum_balance =
+                           Currency.Balance.of_int 1_000_000_000
+                       ; cliff_time = Mina_numbers.Global_slot.of_int 10
+                       ; cliff_amount = Currency.Amount.of_int 1_000_000_000
+                       ; vesting_period = Mina_numbers.Global_slot.of_int 10
+                       ; vesting_increment =
+                           Currency.Amount.of_int 1_000_000_000
+                       }
+                       : Party.Update.Timing_info.value )
+                 in
+                 { Party.Update.dummy with timing } )
+            ; current_auth = Permissions.Auth_required.Signature
+            ; call_data = Snark_params.Tick.Field.zero
+            ; events = []
+            ; sequence_events = []
+            ; protocol_state_precondition = None
+            ; account_precondition = None
+            }
+          in
+          let open Async.Deferred.Let_syntax in
+          let%map update_timing_parties =
+            Transaction_snark.For_tests.update_states ~constraint_constants
+              update_timing_spec
+          in
+          let gen =
+            Quickcheck.Generator.return
+              (ledger_init_state, update_timing_parties)
+          in
+          Quickcheck.test
+            ~seed:
+              (`Deterministic "zkapp command, invalid update for timed account")
+            ~sexp_of:[%sexp_of: Mina_ledger.Ledger.init_state * Parties.t]
+            ~trials:1 gen ~f:(fun (ledger_init_state, update_timing_parties) ->
+              Mina_ledger.Ledger.with_ephemeral_ledger
+                ~depth:constraint_constants.ledger_depth ~f:(fun ledger ->
+                  Mina_ledger.Ledger.apply_initial_ledger_state ledger
+                    ledger_init_state ;
+                  let state_view =
+                    Transaction_snark_tests.Util.genesis_state_view
+                  in
+                  let result =
```

Comment:
call `apply_zkapp_commands_at_slot` instead (for all tests you added)? It applies both in snark and out of snark